### ISSUE TITLE
Fix spinquant implementation

### ISSUE THREAD

**Issue by jerryzh168** [pytorch/ao#3689]
After https://github.com/pytorch/ao/pull/3679/files spinquant implementation imports from `torchao.testing.model_architectures`, this is not ideal for a feature, I think we should import from transformers instead if possible.

And we also want to make sure the spin quant implementation can work for all transformer models (ideally), that's the place for model definition now. (previous implementation might be only working with gpt-fast llama model implementation)

cc @jcaip

**Comment by namgyu-youn** [comment]
Off-topic, but is SpinQuant widely used? I remember Hadamard rotation matrix has really high dequantization overhead without custom kernel impl, and https://huggingface.co/models?search=spinquant seems only a few use case exist.

**Comment by jcaip** [comment]
cc @namgyu-youn 

Although it's not as popular as GPTQ / Smoothquant I think spinquant (or Hadamard / rotation in general) provides the best acuraccy, I've been using the appendix in https://arxiv.org/pdf/2601.09555 as reference:

<img width="727" height="968" alt="Image" src="https://github.com/user-attachments/assets/8e130526-c112-438f-ac41-acb7eff6b5a3" />

**Comment by namgyu-youn** [comment]
@jcaip oh what I mean here is throughput, not accuracy/perplexity. In my last understanding, Hadamard rotation (SpinQuant/QuaRot) generates high overhead, requiring custom kernel impl, unlike GPTQ/AWQ.

For example, SpinQuant impl in torchao is using `fast_hadamard_transform` as a kernel dependency library

https://github.com/pytorch/ao/blob/3075bb6249d82fd580a839ee7ef0e2d655434d28/torchao/prototype/spinquant/hadamard_utils.py#L29-L30

**Comment by jcaip** [comment]
Yes, I think @danielvegamyhre has been working on better hadamard support for training. 

Daniel, do you think that we could maybe use some of that work for inference as well?
